### PR TITLE
[PDI-16624]-Big Number fields are not correctly aggregated in Model V…

### PR DESCRIPTION
**Pull request by ArtsiomYurhevich**
…iew in DET

### CODE REVIEW

**Comment by ArtsiomYurhevich** [comment]
@mbatchelor , @pamval Could you please take a look?

**Comment by pamval** [comment]
@graimundo @mkambol @kurtwalker please review.

File: pdi-dataservice-client/src/main/java/org/pentaho/di/trans/dataservice/jdbc/ThinResultSetMetaData.java [hunk 1]
```diff
@@ -98,7 +98,7 @@ public int getColumnType( int column ) throws SQLException {
       case ValueMetaInterface.TYPE_DATE:
         return java.sql.Types.TIMESTAMP;
       case ValueMetaInterface.TYPE_BIGNUMBER:
-        return java.sql.Types.DECIMAL;
+        return java.sql.Types.JAVA_OBJECT;
```
Comment (mkambol) [inline pdi-dataservice-client/src/main/java/org/pentaho/di/trans/dataservice/jdbc/ThinResultSetMetaData.java:101]: JAVA_OBJECT?  That seems wrong.  Mapping BIGNUMBER (treated as BigDecimal) ->DECIMAL  is consistent with kettle.

Do you know why the wrong type is being used for aggregation?  Is this really an issue in Mondrian's type handling? (maybe JdbcDialectImpl.getType)

**Comment by ArtsiomYurhevich** [comment]
@mkambol I would say that's a safe workaround for proper handling of bigdecimals when the background connection is pdi dataservice connection. I did it because JdbcDialectImpl.getType doesn't work with bigdecimals. For now the bug is that jdbcDialectImpl.getType interprets decimal type ,which ThinResultSetMetaData returns, as Integer and that's why we get bad data. Initially I fixed that and changed it to Double as you wrote in the comment above .All works fine, but when we have big numbers, double will round the value and it will lose accuracy. Currently in mondrian we have the same behaviour and always treat bigdecimals as double values . You can check it in analyzer using any type of connection. But in case we use Object type we get fully correct value w/o rounding. 
The workaround is safe because it touches only pdi dataservice connection type.
If rounding bigdecimals values when we use dataservice connections is what we can accept I will rework the pr to use double type.
Or we can add a new mondrian.rolap.SqlStatement.Type  BigDecimal and this approach will touch many places but will allow us to use bigdecimal types in mondrian.

**Comment by mkambol** [comment]
DET isn't the only client of dataservices, and big_number -> java_object may not work so nicely with others.
An alternative worth considering-- a dataservice specific dialect in Mondrian would allow mapping DECIMAL to object, without messing with other Mondrian dialects or needing to touch dataservices at all.

**Comment by marcbatchelor-nucleus** [comment]
I agree with Matt - if PRD uses dataservices in a report, I imagine it would lose it's mind receiving a JAVA_OBJECT. Please work this out from a dialect.

**Comment by ArtsiomYurhevich** [comment]
+, Reworking the solution

**Pull request closed without merging**